src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -3,17 +3,22 @@ package is.hail.methods
 import breeze.linalg._
 import breeze.numerics.{sigmoid, sqrt}
 import is.hail.annotations._
+import is.hail.distributedmatrix.DistributedMatrix
 import is.hail.expr._
 import is.hail.stats._
 import is.hail.stats.eigSymD.DenseEigSymD
 import is.hail.utils._
 import is.hail.variant.VariantDataset
-
 import org.apache.commons.math3.analysis.UnivariateFunction
 import org.apache.commons.math3.optim.MaxEval
 import org.apache.commons.math3.optim.nonlinear.scalar.GoalType
 import org.apache.commons.math3.optim.univariate.{BrentOptimizer, SearchInterval, UnivariateObjectiveFunction}
 import org.apache.commons.math3.util.FastMath
+import org.apache.spark.mllib.linalg
+import org.apache.spark.mllib.linalg.distributed.BlockMatrix
+
+//FIXME Should not have to make KinshipMatrix and LDMatrix extend this trait. Do fancy type stuff to make this work.
+trait SimilarityMatrix
```

Comment:
Shouldn't this be called `SymmetricMatrix`? It should probably have a `mat: BlockMatrix` and a `keys: Array[T]` and maybe it should be `SymmatrixKeyedMatrix[T]`?